OpenCascade (OCCT) 8_0_0 not supported

**Describe the bug**
Compiling f3d v.3.5.0 with OpenCascade fails with this error:
```
[ 25% 41/135] /usr/local/libexec/ccache/c++ -DF3D_PLUGIN_OCCT_XCAF -DFMT_SHARED -DGLAD_API_CALL_EXPORT -DHAVE_DRACO -DHAVE_FFMPEG -DHAVE_FREEIMAGE -DHAVE_FREETYPE -DHAVE_GLES2_EXT -DHAVE_OPENGL_EXT -DHAVE_RAPIDJSON -DHAVE_TBB -DHAVE_TK -DHAVE_VTK -DHAVE_XLIB -DOCC_CONVERT_SIGNALS -DSCN_DISABLE_FAST_FLOAT=0 -DSCN_DISABLE_REGEX=0 -DSCN_REGEX_BACKEND=0 -DVTK_OPENGL2_BACKEND -Dkiss_fft_scalar=double -I/wrkdirs/usr/ports/graphics/f3d/work/.build/plugins/occt/module -I/wrkdirs/usr/ports/graphics/f3d/work/f3d-3.5.0/plugins/occt/module -isystem /usr/local/include/OpenCASCADE -isystem /usr/local/include/vtk-9.6 -isystem /usr/local/include/vtk-9.6/vtkfmt -isystem /usr/local/include/vtk-9.6/vtknlohmannjson/include -isystem /usr/local/include/vtk-9.6/vtknlohmannjson -isystem /usr/local/include/vtk-9.6/vtkscn/include -isystem /usr/local/include/vtk-9.6/vtkscn -isystem /usr/local/include/vtk-9.6/token -isystem /usr/local/include/vtk-9.6/vtkkissfft -isystem /usr/local/include/freetype2 -isystem /usr/local/include/vtk-9.6/vtkglad -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include  -DNDEBUG -fPIC -pthread -MD -MT plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o -MF plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o.d -o plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o -c /wrkdirs/usr/ports/graphics/f3d/work/f3d-3.5.0/plugins/occt/module/vtkF3DOCCTReader.cxx
FAILED: [code=1] plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o
/usr/local/libexec/ccache/c++ -DF3D_PLUGIN_OCCT_XCAF -DFMT_SHARED -DGLAD_API_CALL_EXPORT -DHAVE_DRACO -DHAVE_FFMPEG -DHAVE_FREEIMAGE -DHAVE_FREETYPE -DHAVE_GLES2_EXT -DHAVE_OPENGL_EXT -DHAVE_RAPIDJSON -DHAVE_TBB -DHAVE_TK -DHAVE_VTK -DHAVE_XLIB -DOCC_CONVERT_SIGNALS -DSCN_DISABLE_FAST_FLOAT=0 -DSCN_DISABLE_REGEX=0 -DSCN_REGEX_BACKEND=0 -DVTK_OPENGL2_BACKEND -Dkiss_fft_scalar=double -I/wrkdirs/usr/ports/graphics/f3d/work/.build/plugins/occt/module -I/wrkdirs/usr/ports/graphics/f3d/work/f3d-3.5.0/plugins/occt/module -isystem /usr/local/include/OpenCASCADE -isystem /usr/local/include/vtk-9.6 -isystem /usr/local/include/vtk-9.6/vtkfmt -isystem /usr/local/include/vtk-9.6/vtknlohmannjson/include -isystem /usr/local/include/vtk-9.6/vtknlohmannjson -isystem /usr/local/include/vtk-9.6/vtkscn/include -isystem /usr/local/include/vtk-9.6/vtkscn -isystem /usr/local/include/vtk-9.6/token -isystem /usr/local/include/vtk-9.6/vtkkissfft -isystem /usr/local/include/freetype2 -isystem /usr/local/include/vtk-9.6/vtkglad -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include -O2 -pipe -fstack-protector-strong -isystem /usr/local/include -fno-strict-aliasing   -isystem /usr/local/include  -DNDEBUG -fPIC -pthread -MD -MT plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o -MF plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o.d -o plugins/occt/module/CMakeFiles/vtkextOCCT.dir/vtkF3DOCCTReader.cxx.o -c /wrkdirs/usr/ports/graphics/f3d/work/f3d-3.5.0/plugins/occt/module/vtkF3DOCCTReader.cxx
/wrkdirs/usr/ports/graphics/f3d/work/f3d-3.5.0/plugins/occt/module/vtkF3DOCCTReader.cxx:23:10: fatal error: 'Standard_PrimitiveTypes.hxx' file not found
   23 | #include <Standard_PrimitiveTypes.hxx>
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.

```
Actually Standard_PrimitiveTypes.hxx was a header installed by the previous versions of OpenCascade, but it does not exist any more.

**Expected behavior**
Compilation is OK.

**System Information:**

- OS: FreeBSD
- GPU and GPU driver: N/A

**F3D Information**
3.5.0

--- a/.github/workflows/versions.json
+++ b/.github/workflows/versions.json
@@ -19,7 +19,7 @@
     "geotiff": "1.7.4",
     "imath": "v3.2.2",
     "java": "25",
-    "occt": "V7_9_3",
+    "occt": "V8_0_0",
     "openexr": "v3.4.4",
     "openvdb": "v13.0.0",
     "ospray": "v2.12.0",
@@ -62,6 +62,6 @@
     "zlib": "v1.2.13"
   },
   "vtk_commit_sha": "1c58a40fe829d968e24e29a91b3fe8e17e0ba14f",
-  "docker_timestamp": "20260504_0",
+  "docker_timestamp": "20260511_0",
   "global_cache_index": "2"
 }

--- a/plugins/occt/module/vtkF3DOCCTReader.cxx
+++ b/plugins/occt/module/vtkF3DOCCTReader.cxx
@@ -20,9 +20,7 @@
 #include <Quantity_Color.hxx>
 #include <STEPControl_Reader.hxx>
 #include <Standard_Handle.hxx>
-#include <Standard_PrimitiveTypes.hxx>
 #include <Storage_StreamTypeMismatchError.hxx>
-#include <TColgp_Array1OfVec.hxx>
 #include <TopExp_Explorer.hxx>
 #include <TopoDS.hxx>
 #include <TopoDS_Edge.hxx>
@@ -81,7 +79,7 @@ vtkCxxSetSmartPointerMacro(vtkF3DOCCTReader, Stream, vtkResourceStream);
 class vtkF3DOCCTReader::vtkInternals
 {
 #if F3D_PLUGIN_OCCT_XCAF
-  using StyleMap = XCAFPrs_IndexedDataMapOfShapeStyle;
+  using StyleMap = NCollection_IndexedDataMap<TopoDS_Shape, XCAFPrs_Style, TopTools_ShapeMapHasher>;
 #endif
 
 public:
@@ -113,7 +111,7 @@ public:
     vtkNew<vtkCellArray> trianglesCells;
     vtkNew<vtkCellArray> linesCells;
 
-    Standard_Integer shift = 0;
+    int shift = 0;
 
 #if F3D_PLUGIN_OCCT_XCAF
     const StyleMap inheritedStyles = this->CollectInheritedStyles(label, shape);
@@ -121,7 +119,7 @@ public:
 
     /* Mesh the whole shape. This only affect faces, edges have to be handled separately. */
     BRepMesh_IncrementalMesh(shape, this->Parent->GetLinearDeflection(),
-      this->Parent->GetRelativeDeflection(), this->Parent->GetAngularDeflection(), Standard_True);
+      this->Parent->GetRelativeDeflection(), this->Parent->GetAngularDeflection(), true);
 
     if (this->Parent->GetReadWire())
     {
@@ -138,8 +136,7 @@ public:
           edges.push_back(edge);
         }
         BRepMesh_IncrementalMesh(compound, this->Parent->GetLinearDeflection(),
-          this->Parent->GetRelativeDeflection(), this->Parent->GetAngularDeflection(),
-          Standard_True);
+          this->Parent->GetRelativeDeflection(), this->Parent->GetAngularDeflection(), true);
       }
 
       // Add all edges to polydata
@@ -153,11 +150,11 @@ public:
           continue;
         }
 
-        Standard_Integer nbV = poly->NbNodes();
+        int nbV = poly->NbNodes();
 
         // Points
-        const TColgp_Array1OfPnt& aNodes = poly->Nodes();
-        for (Standard_Integer i = 1; i <= nbV; i++)
+        const NCollection_Array1<gp_Pnt>& aNodes = poly->Nodes();
+        for (int i = 1; i <= nbV; i++)
         {
           gp_Pnt pt = aNodes(i).Transformed(location);
           points->InsertNextPoint(pt.X(), pt.Y(), pt.Z());
@@ -215,11 +212,11 @@ public:
       Poly::ComputeNormals(poly);
       TopAbs_Orientation faceOrientation = face.Orientation();
 
-      Standard_Integer nbT = poly->NbTriangles();
-      Standard_Integer nbV = poly->NbNodes();
+      int nbT = poly->NbTriangles();
+      int nbV = poly->NbNodes();
 
       // Points
-      for (Standard_Integer i = 1; i <= nbV; i++)
+      for (int i = 1; i <= nbV; i++)
       {
         gp_Pnt pt = poly->Node(i).Transformed(location);
         points->InsertNextPoint(pt.X(), pt.Y(), pt.Z());
@@ -228,7 +225,7 @@ public:
       // Normals
       if (poly->HasNormals())
       {
-        for (Standard_Integer i = 1; i <= nbV; i++)
+        for (int i = 1; i <= nbV; i++)
         {
           gp_Dir n = poly->Normal(i);
           float fn[3] = { static_cast<float>(n.X()), static_cast<float>(n.Y()),
@@ -244,7 +241,7 @@ public:
       {
         // just in case a face does not have normals, add a dummy normal
         float fn[3] = { 0.0, 0.0, 1.0 };
-        for (Standard_Integer i = 1; i <= nbV; i++)
+        for (int i = 1; i <= nbV; i++)
         {
           normals->InsertNextTypedTuple(fn);
         }
@@ -253,7 +250,7 @@ public:
       // UVs
       if (poly->HasUVNodes())
       {
-        for (Standard_Integer i = 1; i <= nbV; i++)
+        for (int i = 1; i <= nbV; i++)
         {
           gp_Pnt2d uv = poly->UVNode(i);
           float fn[2] = { static_cast<float>(uv.X()), static_cast<float>(uv.Y()) };
@@ -576,7 +573,7 @@ public:
   }
 
 protected:
-  void Show(const Message_ProgressScope&, const Standard_Boolean) override
+  void Show(const Message_ProgressScope&, bool) override
   {
     double currentPosition = this->GetPosition();
     if (currentPosition - this->LastPosition > 0.01)
@@ -767,12 +764,12 @@ int vtkF3DOCCTReader::RequestData(
 
   this->Internals->ShapeTool = XCAFDoc_DocumentTool::ShapeTool(doc->Main());
 
-  TDF_LabelSequence topLevelShapes;
+  NCollection_Sequence<TDF_Label> topLevelShapes;
 
   // create polydata leaves
   this->Internals->ShapeTool->GetShapes(topLevelShapes);
 
-  for (Standard_Integer iLabel = 1; iLabel <= topLevelShapes.Length(); ++iLabel)
+  for (int iLabel = 1; iLabel <= topLevelShapes.Length(); ++iLabel)
   {
     TDF_Label label = topLevelShapes.Value(iLabel);
 
@@ -791,7 +788,7 @@ int vtkF3DOCCTReader::RequestData(
 
   vtkNew<vtkMatrix4x4> mat;
   mat->Identity();
-  for (Standard_Integer iLabel = 1; iLabel <= topLevelShapes.Length(); ++iLabel)
+  for (int iLabel = 1; iLabel <= topLevelShapes.Length(); ++iLabel)
   {
     this->Internals->AddLabel(topLevelShapes.Value(iLabel), mat, output);
   }
@@ -832,7 +829,7 @@ int vtkF3DOCCTReader::RequestData(
     ProgressIndicator pi(this);
     reader->TransferRoots(pi.Start());
 
-    Standard_Integer nbShapes = reader->NbShapes();
+    int nbShapes = reader->NbShapes();
 
     output->SetNumberOfBlocks(nbShapes);
 

